### PR TITLE
Move apply_boolean_mask benchmark to nvbench

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Reworks the `apply_booleam_mask` benchmark as an nvbench benchmark under the `STREAM_COMPACTION_NVBENCH` module. `cudf::string_view` was added as a type to help measure the performance improvement in a follow on PR for `apply_boolean_mask` for strings

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/benchmarks/stream_compaction/apply_boolean_mask.cpp [hunk 1]
```diff
@@ -15,120 +15,79 @@
...
-  int64_t const validity_bytes_in  = (fraction >= 1.0f / 32)
-                                       ? cudf::bitmask_allocation_size_bytes(column_size)
-                                       : 4 * column_size_out;
-  int64_t const validity_bytes_out = cudf::bitmask_allocation_size_bytes(column_size_out);
-  int64_t const column_bytes_out   = sizeof(T) * column_size_out;
+  auto const n_rows       = static_cast<cudf::size_type>(state.get_int64("rows"));
+  auto const n_cols       = static_cast<cudf::size_type>(state.get_int64("columns"));
+  auto const percent_true = static_cast<cudf::size_type>(state.get_int64("hits"));
+
+  float const fraction              = percent_true / 100.f;
```
Comment (bdice) [inline cpp/benchmarks/stream_compaction/apply_boolean_mask.cpp:34]: I'd prefer a `double` here. Other benchmarks use `double` for purposes like this.

https://github.com/rapidsai/cudf/blob/115ddcef6451ec7befad69affdafd6a2c8304660/cpp/benchmarks/io/text/multibyte_split.cpp#L121

https://github.com/rapidsai/cudf/blob/115ddcef6451ec7befad69affdafd6a2c8304660/cpp/benchmarks/io/text/multibyte_split.cpp#L131

https://github.com/rapidsai/cudf/blob/115ddcef6451ec7befad69affdafd6a2c8304660/cpp/benchmarks/copying/shift.cu#L59

**Review by bdice — CHANGES_REQUESTED** [review]

File: cpp/benchmarks/stream_compaction/apply_boolean_mask.cpp [hunk 2]
```diff
@@ -15,120 +15,79 @@
...
+  auto const n_cols = static_cast<cudf::size_type>(state.get_int64("columns"));
 
-  data_profile profile = data_profile_builder().cardinality(0).null_probability(0.0).distribution(
-    cudf::type_to_id<T>(), distribution_id::UNIFORM, 0, 100);
+  auto const percent_true = static_cast<cudf::size_type>(state.get_int64("hits"));
 
-  auto source_table = create_random_table(
-    cycle_dtypes({cudf::type_to_id<T>()}, num_columns), row_count{column_size}, profile);
+  auto const input_type = cudf::type_to_id<DataType>();
+  data_profile profile  = data_profile_builder().cardinality(0).null_probability(0.).distribution(
```
Comment (vuule) [inline cpp/benchmarks/stream_compaction/apply_boolean_mask.cpp:64]: just to confirm, we want a nullable column with no nulls here?

File: cpp/benchmarks/stream_compaction/apply_boolean_mask.cpp [hunk 3]
```diff
@@ -15,120 +15,79 @@
...
-BENCHMARK_REGISTER_F(ApplyBooleanMask, double_1_col)->Args({tenM, fifty_percent});
+// using data_type = nvbench::type_list<int8_t, int16_t, int32_t, int64_t, double,
+// cudf::string_view>;
+using data_type = nvbench::type_list<int32_t, int64_t, double, cudf::string_view>;
+NVBENCH_BENCH_TYPES(apply_boolean_mask_benchmark, NVBENCH_TYPE_AXES(data_type))
+  .set_name("apply_boolean_mask")
+  .set_type_axes_names({"type"})
+  .add_int64_axis("columns", {1, 4})
+  .add_int64_axis("rows", {100'000, 1'000'000, 10'000'000})
+  .add_int64_axis("hits", {10, 50, 100});
```
[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/benchmarks/stream_compaction/apply_boolean_mask.cpp:93]: This becomes the column name so I was trying to keep short so the output is not so wide.
Would `percent` be ok?

File: cpp/benchmarks/stream_compaction/apply_boolean_mask.cpp (hunk 2 above)
Comment (davidwendt) [reply cpp/benchmarks/stream_compaction/apply_boolean_mask.cpp:64]: Nope. I will change it to `no_validity()`.

File: cpp/benchmarks/stream_compaction/apply_boolean_mask.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (vuule) [reply cpp/benchmarks/stream_compaction/apply_boolean_mask.cpp:91]: Is something like "hit_%" valid? :D
I'm fine with the current name if there's a reason for the brevity. Not a blocking suggestion at all.

[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/benchmarks/stream_compaction/apply_boolean_mask.cpp:91]: I like `hits_%`.

**Review by vuule — APPROVED** [review]

File: cpp/benchmarks/stream_compaction/apply_boolean_mask.cpp [hunk 4]
```diff
@@ -15,120 +15,76 @@
...
-  int64_t const column_bytes_out   = sizeof(T) * column_size_out;
+  auto const n_rows       = static_cast<cudf::size_type>(state.get_int64("rows"));
+  auto const n_cols       = static_cast<cudf::size_type>(state.get_int64("columns"));
+  auto const percent_true = static_cast<cudf::size_type>(state.get_int64("hits_%"));
+
+  double const fraction             = percent_true / 100.0;
+  cudf::size_type const output_size = fraction * n_rows;
+  int64_t const mask_size = sizeof(bool) * n_rows + cudf::bitmask_allocation_size_bytes(n_rows);
+  int64_t const validity_bytes_in =
+    (fraction >= 1.0f / 32) ? cudf::bitmask_allocation_size_bytes(n_rows) : 4 * output_size;
```
Comment (bdice) [inline cpp/benchmarks/stream_compaction/apply_boolean_mask.cpp:38]: Match the `double` type.
```suggestion
    (fraction >= 1.0 / 32) ? cudf::bitmask_allocation_size_bytes(n_rows) : 4 * output_size;
```

**Review by bdice — APPROVED** [review]

**Pull request merged**